lib/registrant.rb

```diff
@@ -0,0 +1,32 @@
+#type error link: https://www.exceptionalcreatures.com/bestiary/TypeError.html
+
+class Registrant
+  attr_accessor :license_data
+  attr_reader :name, :age, :permit
+
+  def initialize(name, age, permit = false)
+    raise TypeError, "Name must be a String" unless name.is_a?(String)
+    raise TypeError, "Age must be an Integer" unless age.is_a?(Integer)
+    raise TypeError, "permit must be a Boolean" unless permit.is_a?(TrueClass) || permit.is_a?(FalseClass)
+
+    @name = name
+    @age = age
+    @permit = permit
+    @license_data = {:written=>false, :license=>false, :renewed=>false}
+
+  end
+
+  def permit?
+    @permit
+  end
+
+  def earn_permit
+    if @permit
+      return @permit
+    else @permit
+      @permit = true
+    end
```

Comment:
If we just do `@permit = true` it will accomplish all this behavior, no need for an if statement here.